src/Microsoft.AspNet.Razor/Generator/ExpressionCodeGenerator.cs

```diff
@@ -30,12 +32,13 @@ public override string ToString()
 
         public override bool Equals(object obj)
         {
-            return obj is ExpressionCodeGenerator;
+            return obj != null &&
```

Comment:
Even the the `Equals` paradigm on these classes is odd, we may want to move this bit to `HybridCodeGenerator` to stay consistent. Don't feel strongly though.